src/EditorFeatures/Core/Implementation/CommentSelection/AbstractCommentSelectionBase.cs

```diff
@@ -0,0 +1,212 @@
+                .ToImmutableList();
+
+            // Apply the text changes.
+            using (var transaction = new CaretPreservingEditTransaction(title, textView, _undoHistoryRegistry, _editorOperationsFactoryService))
+            {
+                document.Project.Solution.Workspace.ApplyTextChanges(document.Id, edits.TextChanges.Distinct(), CancellationToken.None);
+                transaction.Complete();
+            }
+
+            // Convert the tracking spans into snapshot spans for formatting and selection.
+            var trackingSnapshotSpans = trackingSpans.Select(s => CreateSnapshotSpan(subjectBuffer.CurrentSnapshot, s.trackingSpan, s.originalSpan));
+
+            if (trackingSnapshotSpans.Any())
+            {
+                if (edits.ResultOperation == Operation.Uncomment)
+                {
+                    // Format the document only during uncomment operations.  Use second transaction so it can be undone.
+                    using (var transaction = new CaretPreservingEditTransaction(title, textView, _undoHistoryRegistry, _editorOperationsFactoryService))
+                    {
+                        Format(service, subjectBuffer.CurrentSnapshot, trackingSnapshotSpans, CancellationToken.None);
+                        transaction.Complete();
+                    }
+                }
+
+                // Set the selection after the edits have been applied.
+                var spansToSelect = trackingSnapshotSpans.Select(s => new Selection(s));
+                textView.GetMultiSelectionBroker().SetSelectionRange(spansToSelect, spansToSelect.Last());
+            }
+        }
+
+        /// <summary>
+        /// Creates a tracking span for the operation.
+        /// Internal for tests.
+        /// </summary>
+        internal static ITrackingSpan CreateTrackingSpan(Operation operation, ITextSnapshot snapshot, TextSpan textSpan)
+        {
+            var spanTrackingMode = operation == Operation.Comment
+                ? SpanTrackingMode.EdgeInclusive
+                : SpanTrackingMode.EdgeExclusive;
+            return snapshot.CreateTrackingSpan(Span.FromBounds(textSpan.Start, textSpan.End), spanTrackingMode);
+        }
+
+        /// <summary>
+        /// Retrieves the snapshot span from a post edited tracking span.
+        /// Additionally applies any extra modifications to the tracking span post edit.
+        /// Internal for tests.
+        /// </summary>
+        private static SnapshotSpan CreateSnapshotSpan(ITextSnapshot snapshot, ITrackingSpan trackingSpan, CommentTrackingSpan originalSpan)
+        {
+            var snapshotSpan = trackingSpan.GetSpan(snapshot);
+            if (originalSpan.HasPostApplyChanges())
+            {
+                var updatedStart = snapshotSpan.Start.Position + originalSpan.AmountToAddToStart;
+                var updatedEnd = snapshotSpan.End.Position + originalSpan.AmountToAddToEnd;
```

Comment:
instead of AmonutToAddStart and AmountToAddToEnd can you just store two spans?  the larger and smaller ones?

Reply:
Not easily... basically this amount is not applied to the 'pre-edit' tracking span (```CommentTrackingSpan.TrackingTextSpan```), it's applied to the actual ```ITrackingSpan``` (created from ```CommentTrackingSpan.TrackingTextSpan```) only after the edits have been applied and the tracking span adjusts.

Reply:
interesting.  it would be really helful to have an example somewhere showing what this is taking care of.  i.e. what the code was before, where teh selection is, what the results are, and what needs to be expanded post-edit.  I'm really not clicking on what this is here for.  Thanks!